diff -Naur boost_1_41_0_org/libs/thread/src/win32/thread.cpp boost_1_41_0/libs/thread/src/win32/thread.cpp
--- boost_1_41_0_org/libs/thread/src/win32/thread.cpp	Mon Oct 19 11:18:13 2009
+++ boost_1_41_0/libs/thread/src/win32/thread.cpp	Sat Mar  5 10:21:57 2011
@@ -568,7 +568,7 @@
                 current_node->func=func;
                 current_node->value=tss_data;
             }
-            else
+            else if(func && tss_data)
             {
                 detail::thread_data_base* const current_thread_data(get_or_make_current_thread_data());
                 tss_data_node* const new_node=heap_new<tss_data_node>(key,func,tss_data,current_thread_data->tss_data);
