diff -Naur include_orig/CGAL/Arr_accessor.h include/CGAL/Arr_accessor.h
--- include_orig/CGAL/Arr_accessor.h	2008-11-29 10:11:47.000000000 -0500
+++ include/CGAL/Arr_accessor.h	2008-11-29 10:33:23.000000000 -0500
@@ -225,6 +225,9 @@
     return (p_arr->_handle_for (p_arr->v_tr));
   }
 
+  // BAS hack 
+  DFace * un_face() { return p_arr->un_face; }
+
   /*! Get the fictitious face of the arrangement (non-const version). */
   Face_handle fictitious_face ()
   {
diff -Naur include_orig/CGAL/Arr_landmarks_point_location.h include/CGAL/Arr_landmarks_point_location.h
--- include_orig/CGAL/Arr_landmarks_point_location.h	2008-11-29 10:11:47.000000000 -0500
+++ include/CGAL/Arr_landmarks_point_location.h	2008-11-29 10:13:04.000000000 -0500
@@ -158,7 +158,7 @@
     CGAL_assertion(lm_gen != NULL);
     if (lm_gen)
     {
-      lm_gen->detach(p_arr);    
+      lm_gen->detach(/*p_arr*/);    
     }
   }
   
diff -Naur include_orig/CGAL/centroid.h include/CGAL/centroid.h
--- include_orig/CGAL/centroid.h	2008-11-29 10:11:47.000000000 -0500
+++ include/CGAL/centroid.h	2008-11-29 10:12:43.000000000 -0500
@@ -60,7 +60,7 @@
     v = v + (*begin++ - ORIGIN);
     nb_pts++;
   }
-  return ORIGIN + v / (FT)nb_pts;
+  return ORIGIN + v / (FT)(int)nb_pts;
 }// end centroid of a 2D point set
 
 // computes the centroid of a 3D point set
diff -Naur include_orig/CGAL/General_polygon_set_2.h include/CGAL/General_polygon_set_2.h
--- include_orig/CGAL/General_polygon_set_2.h	2008-11-29 10:11:47.000000000 -0500
+++ include/CGAL/General_polygon_set_2.h	2008-11-29 10:12:21.000000000 -0500
@@ -55,7 +55,7 @@
   typedef std::size_t                                     Size;
 
 private:
-  typedef General_polygon_set_2<Traits_>                  Self;
+  typedef General_polygon_set_2<Traits_,Dcel_>            Self;
   typedef typename Traits_2::Point_2                      Point_2;
   typedef typename Traits_2::X_monotone_curve_2           X_monotone_curve_2;
   
