diff -Naur CGAL-3.3.1/include/CGAL/Arr_accessor.h CGAL-3.3.1.new/include/CGAL/Arr_accessor.h
--- CGAL-3.3.1/include/CGAL/Arr_accessor.h	Sat Aug 25 21:01:33 2007
+++ CGAL-3.3.1.new/include/CGAL/Arr_accessor.h	Thu Apr  2 01:07:43 2009
@@ -225,6 +225,9 @@
     return (p_arr->_handle_for (p_arr->v_tr));
   }
 
+  // BAS hack 
+  DFace * un_face() { return p_arr->un_face; }
+
   /*! Get the fictitious face of the arrangement (non-const version). */
   Face_handle fictitious_face ()
   {
diff -Naur CGAL-3.3.1/include/CGAL/Arr_landmarks_point_location.h CGAL-3.3.1.new/include/CGAL/Arr_landmarks_point_location.h
--- CGAL-3.3.1/include/CGAL/Arr_landmarks_point_location.h	Sat Aug 25 21:01:32 2007
+++ CGAL-3.3.1.new/include/CGAL/Arr_landmarks_point_location.h	Thu Apr  2 01:07:43 2009
@@ -158,7 +158,7 @@
     CGAL_assertion(lm_gen != NULL);
     if (lm_gen)
     {
-      lm_gen->detach(p_arr);    
+      lm_gen->detach(/*p_arr*/);    
     }
   }
   
diff -Naur CGAL-3.3.1/include/CGAL/General_polygon_set_2.h CGAL-3.3.1.new/include/CGAL/General_polygon_set_2.h
--- CGAL-3.3.1/include/CGAL/General_polygon_set_2.h	Sat Aug 25 21:01:32 2007
+++ CGAL-3.3.1.new/include/CGAL/General_polygon_set_2.h	Thu Apr  2 01:07:43 2009
@@ -55,7 +55,7 @@
   typedef std::size_t                                     Size;
 
 private:
-  typedef General_polygon_set_2<Traits_>                  Self;
+  typedef General_polygon_set_2<Traits_,Dcel_>            Self;
   typedef typename Traits_2::Point_2                      Point_2;
   typedef typename Traits_2::X_monotone_curve_2           X_monotone_curve_2;
   
diff -Naur CGAL-3.3.1/include/CGAL/centroid.h CGAL-3.3.1.new/include/CGAL/centroid.h
--- CGAL-3.3.1/include/CGAL/centroid.h	Sat Aug 25 21:01:33 2007
+++ CGAL-3.3.1.new/include/CGAL/centroid.h	Thu Apr  2 01:07:43 2009
@@ -60,7 +60,7 @@
     v = v + (*begin++ - ORIGIN);
     nb_pts++;
   }
-  return ORIGIN + v / (FT)nb_pts;
+  return ORIGIN + v / (FT)(int)nb_pts;
 }// end centroid of a 2D point set
 
 // computes the centroid of a 3D point set
diff -Naur CGAL-3.3.1/include/CGAL/triangulation_assertions.h CGAL-3.3.1.new/include/CGAL/triangulation_assertions.h
--- CGAL-3.3.1/include/CGAL/triangulation_assertions.h	Sat Aug 25 21:01:32 2007
+++ CGAL-3.3.1.new/include/CGAL/triangulation_assertions.h	Tue Mar 31 18:51:43 2009
@@ -12,7 +12,7 @@
 // WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
 //
 // $URL: svn+ssh://scm.gforge.inria.fr/svn/cgal/branches/CGAL-3.3-branch/Triangulation_2/include/CGAL/triangulation_assertions.h $
-// $Id$
+// $Id$
 // 
 //
 // Author(s)     : script by Geert-Jan Giezeman and Sven Schoenherr 
@@ -24,7 +24,7 @@
 // assertions
 // ----------
 
-
+#ifndef CGAL_triangulation_assertion
 #if defined(CGAL_TRIANGULATION_NO_ASSERTIONS) || defined(CGAL_NO_ASSERTIONS) \
   || defined(NDEBUG)
 #  define CGAL_triangulation_assertion(EX) (static_cast<void>(0))
@@ -261,5 +261,5 @@
    ((EX)?(static_cast<void>(0)): ::CGAL::warning_fail( # EX , __FILE__, __LINE__, MSG))
 #  define CGAL_triangulation_expensive_exactness_warning_code(CODE) CODE
 #endif // CGAL_TRIANGULATION_NO_WARNINGS
-
+#endif
 
