diff -Naur CGAL-3.4_org/include/CGAL/Constrained_triangulation_2.h CGAL-3.4/include/CGAL/Constrained_triangulation_2.h
--- CGAL-3.4_org/include/CGAL/Constrained_triangulation_2.h	2008-12-19 14:49:08.000000000 +0100
+++ CGAL-3.4/include/CGAL/Constrained_triangulation_2.h	2009-08-27 11:49:21.593561746 +0200
@@ -370,7 +370,7 @@
   Face_handle loc;
   int li;
   Locate_type lt;
-  loc = locate(a, lt, li, start);
+  loc = Constrained_triangulation::locate(a, lt, li, start);
   return Constrained_triangulation::insert(a,lt,loc,li);
 }
 
diff -Naur CGAL-3.4_org/include/CGAL/Convex_hull_2/ch_assertions.h CGAL-3.4/include/CGAL/Convex_hull_2/ch_assertions.h
--- CGAL-3.4_org/include/CGAL/Convex_hull_2/ch_assertions.h	2008-12-19 14:48:26.000000000 +0100
+++ CGAL-3.4/include/CGAL/Convex_hull_2/ch_assertions.h	2009-08-27 11:52:04.281560389 +0200
@@ -17,7 +17,8 @@
 //
 // Author(s)     : script by Geert-Jan Giezeman and Sven Schoenherr 
 
-
+#ifndef CH_ASSERTIONS_H
+#define CH_ASSERTIONS_H
 
 
 // macro definitions
@@ -263,4 +264,5 @@
 #  define CGAL_ch_expensive_exactness_warning_code(CODE) CODE
 #endif // CGAL_CH_NO_WARNINGS
 
+#endif /* CH_ASSERTIONS_H */
 
diff -Naur CGAL-3.4_org/include/CGAL/General_polygon_set_2.h CGAL-3.4/include/CGAL/General_polygon_set_2.h
--- CGAL-3.4_org/include/CGAL/General_polygon_set_2.h	2008-12-19 14:50:31.000000000 +0100
+++ CGAL-3.4/include/CGAL/General_polygon_set_2.h	2009-08-27 11:49:55.329375604 +0200
@@ -55,7 +55,7 @@
   typedef std::size_t                                     Size;
 
 private:
-  typedef General_polygon_set_2<Traits_>                  Self;
+  typedef General_polygon_set_2<Traits_, Dcel_>                  Self;
   typedef typename Traits_2::Point_2                      Point_2;
   typedef typename Traits_2::X_monotone_curve_2           X_monotone_curve_2;
   
diff -Naur CGAL-3.4_org/include/CGAL/Polygon_2/polygon_assertions.h CGAL-3.4/include/CGAL/Polygon_2/polygon_assertions.h
--- CGAL-3.4_org/include/CGAL/Polygon_2/polygon_assertions.h	2008-12-19 14:50:03.000000000 +0100
+++ CGAL-3.4/include/CGAL/Polygon_2/polygon_assertions.h	2009-08-27 11:52:59.330894985 +0200
@@ -21,6 +21,8 @@
 //
 // Author(s)     : script by Geert-Jan Giezeman and Sven Schoenherr 
 
+#ifndef POLYGON_ASSERTIONS_H
+#define POLYGON_ASSERTIONS_H
 
 // macro definitions
 // =================
@@ -265,3 +267,5 @@
 #  define CGAL_polygon_expensive_exactness_warning_code(CODE) CODE
 #endif // CGAL_POLYGON_NO_WARNINGS
 
+#endif /* POLYGON_ASSERTIONS_H */
+
diff -Naur CGAL-3.4_org/include/CGAL/triangulation_assertions.h CGAL-3.4/include/CGAL/triangulation_assertions.h
--- CGAL-3.4_org/include/CGAL/triangulation_assertions.h	2008-12-19 14:49:08.000000000 +0100
+++ CGAL-3.4/include/CGAL/triangulation_assertions.h	2009-08-27 11:50:57.537214527 +0200
@@ -17,7 +17,8 @@
 //
 // Author(s)     : script by Geert-Jan Giezeman and Sven Schoenherr 
 
-
+#ifndef TRIANGULATION_ASSERTIONS_H
+#define TRIANGULATION_ASSERTIONS_H
 
 // macro definitions
 // =================
@@ -262,4 +263,5 @@
 #  define CGAL_triangulation_expensive_exactness_warning_code(CODE) CODE
 #endif // CGAL_TRIANGULATION_NO_WARNINGS
 
+#endif /* TRIANGULATION_ASSERTIONS_H */
 
diff -Naur CGAL-3.4_org/include/CGAL/Delaunay_triangulation_2.h CGAL-3.4/include/CGAL/Delaunay_triangulation_2.h
--- CGAL-3.4_org/include/CGAL/Delaunay_triangulation_2.h	2008-12-19 14:49:08.000000000 +0100
+++ CGAL-3.4/include/CGAL/Delaunay_triangulation_2.h	2009-08-27 11:50:57.537214527 +0200
@@ -533,7 +533,7 @@
 {
   Locate_type lt;
   int li;
-  Face_handle loc = locate (p, lt, li, start);
+  Face_handle loc = Triangulation::locate (p, lt, li, start);
   return insert(p, lt, loc, li);
 }
   
